modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -0,0 +1,246 @@
+) -> np.ndarray:
+    """
+    Determine quantiles from the given samples.
+
+    This function takes as input the quantiles calculated over all partitions from
+    `sample_func` defined above, and determines a final NPartitions.get() * 2 quantiles
+    to use to roughly sort the entire dataframe. It does so by collating all the samples
+    and computing NPartitions.get() * 2 quantiles for the overall set.
+
+    Parameters
+    ----------
+    df : PandasDataframe
+        The Modin Dataframe calling this function.
+    samples : np.ndarray
+        The samples computed by ``get_partition_quantiles_for_sort``.
+    columns : list[str]
+        The columns to sort by.
+    method : str, default: linear
+        The method to use when picking quantiles.
+
+    Returns
+    -------
+    np.ndarray
+        A list of overall quantiles.
+    """
+    # We don't call `np.unique` on the samples, since if a quantile shows up in multiple
+    # partition's samples, this is probably an indicator of skew in the dataset, and we
+    # want our final partitions to take this into account.
+    all_pivots = np.array(samples).flatten()
+    quantiles = [i / (NPartitions.get() * 2) for i in range(NPartitions.get() * 2)]
+    overall_quantiles = np.quantile(all_pivots, quantiles, method=method)
+    if df.dtypes[columns[0]] != object:
+        overall_quantiles[0] = np.NINF
+        overall_quantiles[-1] = np.inf
+    return overall_quantiles
+
+
+def split_partitions_using_pivots_for_sort(
+    modin_frame: "PandasDataframe",
+    df: pandas.DataFrame,
+    columns: list,
+    pivots: np.ndarray,
+    ascending: Union[list, bool],
+    **kwargs: dict
+) -> "tuple[pandas.DataFrame]":
+    """
+    Split the given dataframe into the partitions specified by `pivots`.
+
+    This function takes as input a row-axis partition, as well as the quantiles determined
+    by the `pivot_func` defined above. It then splits the input dataframe into NPartitions.get() * 2
+    dataframes, with the elements in the i-th split belonging to the i-th partition, as determined
+    by the quantiles we're using.
+
+    Parameters
+    ----------
+    modin_frame : PandasDataframe
+        The Modin Dataframe calling this function.
+    df : pandas.Dataframe
+        The partition to split.
+    columns : list[str]
+        The columns to sort by.
+    pivots : np.ndarray
+        The quantiles to use to split the data.
+    ascending : list[bool] or bool
+        The ascending flag (or a list of ascending flags for each column).
+    **kwargs : dict
+        Additional keyword arguments.
+
+    Returns
+    -------
+    list[pandas.DataFrame]
+        A list of the splits from this partition.
+    """
+    if isinstance(ascending, list):
+        ascending = ascending[0]
+    if not ascending and modin_frame.dtypes[columns[0]] != object:
+        pivots = pivots[::-1]
+    na_rows = df[df[columns[0]].isna()]
```

Comment:
Could you please save `columns[0]` instead of repeatedly getting it?

Reply:
Done!